File: locator/src/main/java/com/printf/shuttleTracker/locationServices/FirebaseSender.java
package com.printf.shuttleTracker.locationServices;

import android.util.Log;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class FirebaseSender {
    private final FirebaseDatabase database = FirebaseDatabase.getInstance();
    public DatabaseReference latitudeContainer = database.getReference("commericalLocationA").child("latitude");
    public DatabaseReference longitudeContainer = database.getReference("commericalLocationA").child("longitude");
    public DatabaseReference activeContainer = database.getReference("commericalLocationA").child("isActive");

    FirebaseSender(double latitude, double longitude, boolean isActive){
        latitudeContainer.setValue(latitude);
        longitudeContainer.setValue(longitude);
        activeContainer.setValue(isActive, (error, ref) -> {
            if(error != null){
            Log.d("Firebase Error", String.valueOf(error));
        }});
    }




}

File: locator/src/main/java/com/printf/shuttleTracker/locationServices/FirebaseSender.kt
package com.printf.shuttleTracker.locationServices

import android.util.Log
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.DatabaseError

class FirebaseSender internal constructor(latitude: Double, longitude: Double, isActive: Boolean) {
    private val database = FirebaseDatabase.getInstance()
    var latitudeContainer = database.getReference("commericalLocationA").child("latitude")
    var longitudeContainer = database.getReference("commericalLocationA").child("longitude")
    var activeContainer = database.getReference("commericalLocationA").child("isActive")

    init {
        latitudeContainer.setValue(latitude)
        longitudeContainer.setValue(longitude)
        activeContainer.setValue(isActive) { error: DatabaseError?, ref: DatabaseReference? ->
            if (error != null) {
                Log.d("Firebase Error", error.toString())
            }
        }
    }
}